pkg/azuredevops/smoke/smoketests.go

```diff
@@ -0,0 +1,401 @@
+			if strings.Contains(line, logFinding) {
+				return true, nil
+			}
+		}
+	}
+
+	return false, nil
+}
+
+func checkSpecificBuildForMissingCommand(ctx context.Context, connection *azuredevops.Connection, projectName, pipelineName, expectedMissingMessage string, pipelineID int) (bool, error) {
+	builds, err := getSpecificBuilds(ctx, connection, projectName, pipelineID)
+	if err != nil {
+		return false, fmt.Errorf("error getting last build: %w", err)
+	}
+	if len(builds) == 0 {
+		return false, fmt.Errorf("no builds found for pipeline %s", pipelineName)
+	}
+
+	buildClient, err := build.NewClient(ctx, connection)
+	if err != nil {
+		return false, fmt.Errorf("error creating build client: %w", err)
+	}
+
+	lastBuildID := builds[0].Id
+	logs, err := buildClient.GetBuildLogs(ctx, build.GetBuildLogsArgs{
+		Project: &projectName,
+		BuildId: lastBuildID,
+	})
+	if err != nil {
+		return false, fmt.Errorf("error getting build logs: %w", err)
+	}
+
+	// Search logs for usage of command `***`
+	for _, buildLog := range *logs {
+		logContent, err := buildClient.GetBuildLogLines(ctx, build.GetBuildLogLinesArgs{
+			Project: &projectName,
+			BuildId: lastBuildID,
+			LogId:   buildLog.Id,
+		})
+		if err != nil {
+			return false, fmt.Errorf("error getting build buildLog lines: %w", err)
+		}
+
+		for _, line := range *logContent {
+			if strings.Contains(line, expectedMissingMessage) {
+				return false, fmt.Errorf("unexpected message found in logs: %s", expectedMissingMessage)
+			}
+		}
+	}
+
+	return true, nil
+}
+
+func RunBuildTest(ctx context.Context, connection *azuredevops.Connection, projectName, pipelineName string, pipelineID int, test BuildTest) bool {
+	var pass bool
+	var err error
+
+	if test.ExpectAbsent {
+		pass, err = checkSpecificBuildForMissingCommand(ctx, connection, projectName, pipelineName, test.LogMessage, pipelineID)
+	} else {
+		pass, err = checkSpecificBuildForCommand(ctx, connection, projectName, pipelineName, test.LogMessage, pipelineID)
+	}
+
+	if err != nil {
+		log.Fatalf("Test failed for %s: %v\n", test.Description, err)
+	}
+
+	if !pass {
+		log.Fatalf("Test failed for %s: condition not met\n", test.Description)
+	}
+
+	fmt.Printf("Test passed for %s\n", test.Description)
+	return true
+}
+
+func RunTimelineTests(ctx context.Context, connection *azuredevops.Connection, projectName string, buildID int, test TimelineTest) bool {
+	var pass bool
+	var err error
+
+	pass, err = getBuildStageStatus(ctx, connection, projectName, buildID, test)
+	if err != nil {
+		log.Fatalf("Test failed for %s: %v\n", test.Name, err)
+	}
+
+	if !pass {
+		log.Fatalf("Test failed for %s: condition not met\n", test.Name)
+	}
+
+	fmt.Printf("Test passed for %s\n", test.Name)
+	return true
+}
+
+func ShouldRunTest(testsToRun string, testsToRunList []string, testName string) bool {
+	if testsToRun == "all" || testsToRun == "" {
+		return true
+	}
+	return slices.Contains(testsToRunList, testName)
+}
+
+func GetBuildTests() []BuildTest {
```

Comment:
BuildTest definitions should be maintained in separate file as an yaml or json. This way we can maintain definitions in separate from a tool, even have it in more than one location if that would make sense for us.